renku/core/commands/init.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Project initialization logic."""
+
+import tempfile
+from datetime import datetime, timezone
+from pathlib import Path
+
+import git
+
+from renku.core.management.config import RENKU_HOME
+
+
+def fetch_remote_template(
+    url, folder, branch='master', tempdir=Path(tempfile.mkdtemp())
+):
+    """Fetch the remote template and checkout the relevant folder.
+
+    Returns:
+        The path with template files.
+
+    """
+    # clone the repo locally without checking out files
+    template_repo = git.Repo.clone_from(
+        url, tempdir, no_checkout=True, depth=1
+    )
+
+    # check branch
+    template_branches = [branch.name for branch in template_repo.branches]
```

Comment:
you could use a set comprehension here instead for faster lookup

Reply:
Changed it, I hope it's in line with what you had in mind. Please let me know if there is an obvious better way to do that